docs/src/main/tut/freeapplicative.md

```diff
@@ -5,10 +5,167 @@ section: "data"
+
+```tut
+import cats.Id
+import cats.arrow.NaturalTransformation
+import cats.std.function._
+
+val compiler =
+  new NaturalTransformation[ValidationOp, String => ?] {
+    def apply[A](fa: ValidationOp[A]): String => A =
+      str =>
+        fa match {
+          case Size(size) => str.size >= size
+          case HasNumber  => str.exists(c => "0123456789".contains(c))
+        }
+  }
+
+val validator = prog.foldMap[String => ?](compiler)
+validator("1234")
+validator("12345")
+```
+
+## Differences from `Free`
+So far everything we've been doing has been not much different from `Free` - we've built
+an algebra and interpreted it. However, there are some things `FreeApplicative` can do that
+`Free` cannot.
+
+Recall a key distinction between the type classes `Applicative` and `Monad` - `Applicative`
+captures the idea of independent computations, whereas `Monad` captures that of dependent
+computations. Put differently `Applicative`s cannot branch based on the value of an existing/prior
+computation. Therefore when using `Applicative`s, we must hand in all our data in one go.
+
+In the context of `FreeApplicative`s, we can leverage this static knowledge in our interpreter.
+
+### Parallelism
+Because we have everything we need up front and know there can be no branching, we can easily
+write a validator that validates in parallel.
+
+```tut:silent
+import cats.data.Kleisli
+import cats.std.future._
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+// recall Kleisli[Future, String, A] is the same as String => Future[A]
+type ParValidator[A] = Kleisli[Future, String, A]
+
+val parCompiler =
+  new NaturalTransformation[ValidationOp, ParValidator] {
+    def apply[A](fa: ValidationOp[A]): ParValidator[A] =
+      Kleisli { str =>
+        fa match {
+          case Size(size) => Future { str.size >= size }
+          case HasNumber  => Future { str.exists(c => "0123456789".contains(c)) }
+        }
+      }
+  }
+
+val parValidation = prog.foldMap[ParValidator](parCompiler)
+```
+
+### Logging
+We can also write an interpreter that simply creates a list of strings indicating the filters that
+have been used - this could be useful for logging purposes. Note that we need not actually evaluate
+the rules against a string for this, we simply need to map each rule to some identifier. Therefore
+we can completely ignore the return type of the operation and return just a `List[String]` - the
+`Const` data type is useful for this.
+
+```tut:silent
+import cats.data.Const
+import cats.std.list._
+
+type Log[A] = Const[List[String], A]
+
+val logCompiler =
+  new NaturalTransformation[ValidationOp, Log] {
+    def apply[A](fa: ValidationOp[A]): Log[A] =
+      fa match {
+        case Size(size) => Const(List(s"size >= $size"))
+        case HasNumber  => Const(List("has number"))
+      }
+  }
+
+val logValidation = prog.foldMap[Log](logCompiler)
```

Comment:
It might be nice to put a non-silent tut block under this one that feeds some input in and shows the output.